src/Particle/WalkerConfigurations.h

```diff
@@ -0,0 +1,201 @@
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2020 QMCPACK developers.
+//
+// File developed by: Jordan E. Vincent, University of Illinois at Urbana-Champaign
+//                    Ken Esler, kpesler@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jeremy McMinnis, jmcminis@gmail.com, University of Illinois at Urbana-Champaign
+//                    Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
+//                    Cynthia Gu, zg1@ornl.gov, Oak Ridge National Laboratory
+//                    Raymond Clay III, j.k.rofling@gmail.com, Lawrence Livermore National Laboratory
+//                    Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//                    Mark A. Berrill, berrillma@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+/** @file WalkerConfigurations.h
+ * @brief Declaration of a WalkerConfigurations
+ */
+#ifndef QMCPLUSPLUS_WALKERCONFIGURATIONS_H
+#define QMCPLUSPLUS_WALKERCONFIGURATIONS_H
+#include "Configuration.h"
+#include "Particle/Walker.h"
+#include "Utilities/IteratorUtility.h"
+
+namespace qmcplusplus
+{
+/** Monte Carlo Data of an ensemble
+ *
+ * The quantities are shared by all the nodes in a group
+ * - NumSamples number of samples
+ * - Weight     total weight of a sample
+ * - Energy     average energy of a sample
+ * - Variance   variance
+ * - LivingFraction fraction of walkers alive each step.
+ */
+template<typename T>
+struct MCDataType
+{
+  T NumSamples;
+  T RNSamples;
+  T Weight;
+  T Energy;
+  T AlternateEnergy;
+  T Variance;
+  T R2Accepted;
+  T R2Proposed;
+  T LivingFraction;
+};
+
+/** A set of light weight walkers that are carried between driver sections and restart
+ */
+class WalkerConfigurations
+{
+public:
+  /// walker type
+  using Walker_t         = Walker<QMCTraits, PtclOnLatticeTraits>;
+  using FullPrecRealType = QMCTraits::FullPrecRealType;
+  ///container type of Walkers
+  using WalkerList_t = std::vector<Walker_t*>;
+  /// FIX: a type alias of iterator for an object should not be for just one of many objects it holds.
+  using iterator = WalkerList_t::iterator;
+  ///const_iterator of Walker container
+  using const_iterator = WalkerList_t::const_iterator;
+
+  /** starting index of the walkers in a processor group
+   *
+   * WalkerOffsets[0]=0 and WalkerOffsets[WalkerOffsets.size()-1]=total number of walkers in a group
+   * WalkerOffsets[processorid+1]-WalkerOffsets[processorid] is equal to the number of walkers on a processor,
+   * i.e., W.getActiveWalkers().
+   * WalkerOffsets is added to handle parallel I/O with hdf5
+   */
+  std::vector<int> WalkerOffsets;
```

Comment:
The idea of "offsets" is not used in the new population control. But that can be dealt with when this is used with the unified driver.